### PR TITLE
Implement a Superposition Interface

### PR DESCRIPTION
**Pull request by lafita**
This PR addresses #558, with the following changes:

- Implement a `SuperPosition` interface to unify the three superposition algorithms in **biojava**.
- Implement a `SuperPositions` Class with a singleton superposition object, so that the default superposition algorithm used in **biojava** can be easily changed and to provide an easy access to single superpositions.
- Move some methods in `Calc` operating on transformation and rotation matrices to a new Class `Matrices`.

### CODE REVIEW

File: biojava-structure/src/main/java/org/biojava/nbio/structure/geometry/SuperPositionQCP.java [hunk 1]
```diff
@@ -107,20 +107,13 @@
...
-	 * The required eigenvector precission
-	 */
-	private static final double EVEC_PREC = 1E-6;
-	
-	/**
-	 * The required eigenvalue precission
-	 */
-	private static final double EVAL_PREC = 1E-11;
+	private double EVEC_PREC = 1E-6;
+	private double EVAL_PREC = 1E-11;
```
Comment (josemduarte) [inline biojava-structure/src/main/java/org/biojava/nbio/structure/geometry/SuperPositionQCP.java]: These 2 should be lower case if they are variable fields

**Comment by josemduarte** [comment]
This looks fantastic! 

Should we close PR #576 then? If I understand it, this PR would solve the issue discussed there, wouldn't it?

**Comment by andreasprlic** [comment]
#576 is a small patch of alignment tools. I can redo this later once this has been merged in.

File: biojava-structure/src/main/java/org/biojava/nbio/structure/geometry/SuperPositionQCP.java (hunk 1 above)
Comment (lafita) [inline biojava-structure/src/main/java/org/biojava/nbio/structure/geometry/SuperPositionQCP.java]: Fixed! Thanks

**Comment by lafita** [comment]
I think we can close #545 with this PR. The only geometry issue that has still to be addressed is #303.

**Comment by andreasprlic** [comment]
can we make AtomImpl use an internal Point3D to store coords?

**Comment by josemduarte** [comment]
I'll merge this. I've opened a new issue for the AtomImpl thing #582

**Pull request merged**